jetbot/bot.py

```diff
@@ -69,7 +69,42 @@ def product_url(self, data: str) -> str:
     def issue_url(self, data: str) -> str:
         return "<https://youtrack.jetbrains.com/issues/" + data + ">"
 
-    # Find a product channel
+    # Find product channels
+    async def product_channels(self, data: dict) -> list[TextChannel]:
+        channels = []
+        if not data.get("category_name"):
+            return channels
+
+        guild = self.get_guild(self.config.guild)
+        if not guild:
+            guild = await self.fetch_guild(self.config.guild)
+
+        text_channels = guild.text_channels
+        if not text_channels:
+            text_channels = await guild.fetch_channels()
+            text_channels = [f for f in text_channels if isinstance(f, TextChannel)]
+
+        # Handle multi-channel configuration
+        if "channels" in data:
+            for channel_config in data["channels"]:
+                for channel in text_channels:
+                    if channel.name == channel_config["name"] \
+                            and channel.category \
+                            and channel.category.name.lower().strip() == data["category_name"].lower().strip():
+                        channels.append(channel)
+                        break
+        # Handle legacy single-channel configuration
+        elif data.get("channel_name"):
+            for channel in text_channels:
+                if channel.name == data["channel_name"] \
+                        and channel.category \
+                        and channel.category.name.lower().strip() == data["category_name"].lower().strip():
+                    channels.append(channel)
+                    break
+
+        return channels
+
+    # Find a product channel (legacy support)
```

Comment:
Do we need this legacy method, or can we update everything to handle multiple channels from the method above?

Reply:
I would keep it around for now in case we have a new product which will use the old structure for now (which is fairly likely going to happen :D)

Reply:
Right, but new products with a single channel should still work with the new method?

Reply:
ohhh I see what you are saying, yeah that's true that makes sense, this can probably be generalized. Will look into it!